src/cs_transmog.cpp

```diff
@@ -56,6 +68,160 @@ class transmog_commandscript : public CommandScript
+            player = PlayerIdentifier::FromTargetOrSelf(handler);
+        if (!player || !player->IsConnected())
+            return false;
+
+        Player* target = player->GetConnectedPlayer();
+
+        if (!sTransmogrification->GetTrackUnusableItems() && !sTransmogrification->SuitableForTransmogrification(target, itemTemplate))
+        {
+            handler->SendSysMessage(LANG_CMD_TRANSMOG_ADD_UNSUITABLE);
+            handler->SetSentErrorMessage(true);
+            return true;
+        }
+
+        if (itemTemplate->Class != ITEM_CLASS_ARMOR && itemTemplate->Class != ITEM_CLASS_WEAPON)
+        {
+            handler->SendSysMessage(LANG_CMD_TRANSMOG_ADD_FORBIDDEN);
+            handler->SetSentErrorMessage(true);
+            return true;
+        }
+
+        uint32 itemId = itemTemplate->ItemId;
+        uint32 accountId = target->GetSession()->GetAccountId();
+        std::stringstream tempStream;
+        tempStream << std::hex << ItemQualityColors[itemTemplate->Quality];
+        std::string itemQuality = tempStream.str();
+        std::string itemName = itemTemplate->Name1;
+        std::string nameLink = handler->playerLink(target->GetName());
+        if (sTransmogrification->AddCollectedAppearance(accountId, itemId))
+        {
+
+            if (!(target->GetPlayerSetting("mod-transmog", SETTING_HIDE_TRANSMOG).value))
+            {
+                ChatHandler(target->GetSession()).PSendSysMessage(R"(|c%s|Hitem:%u:0:0:0:0:0:0:0:0|h[%s]|h|r has been added to your appearance collection.)", itemQuality.c_str(), itemId, itemName.c_str());
+            }
+
+            if (target != handler->GetPlayer())
+            {
+                handler->PSendSysMessage(R"(|c%s|Hitem:%u:0:0:0:0:0:0:0:0|h[%s]|h|r has been added to the appearance collection of Player %s.)", itemQuality.c_str(), itemId, itemName.c_str(), nameLink);
+            }
+
+            CharacterDatabase.Execute("INSERT INTO custom_unlocked_appearances (account_id, item_template_id) VALUES ({}, {})", accountId, itemId);
+        }
+        else
+        {
+            handler->PSendSysMessage(R"(Player %s already has item |c%s|Hitem:%u:0:0:0:0:0:0:0:0|h[%s]|h|r in the appearance collection.)", nameLink, itemQuality.c_str(), itemId, itemName.c_str());
+            handler->SetSentErrorMessage(true);
+        }
+
+        return true;
+    }
+
+    static bool HandleAddTransmogItemSet(ChatHandler* handler, Optional<PlayerIdentifier> player, Variant<Hyperlink<itemset>, uint32> itemSetId)
+    {
+        if (!sTransmogrification->GetUseCollectionSystem())
+            return true;
+
+        if (!*itemSetId)
+        {
+            handler->PSendSysMessage(LANG_NO_ITEMS_FROM_ITEMSET_FOUND, uint32(itemSetId));
+            handler->SetSentErrorMessage(true);
+            return false;
+        }
+
+        if (!player)
+            player = PlayerIdentifier::FromTargetOrSelf(handler);
+        if (!player || !player->IsConnected())
+            return false;
```

Comment:
you could make the same changes of above here, moreover this code is similar to the other one above, you could create a private function that wrap all the common code